lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1355,6 +1345,44 @@ def compute_timestamp(resource_type, record, time)
         ts_secs = timestamp.tv_sec
         ts_nanos = timestamp.tv_nsec
       end
+      ts_secs = begin
+                  Integer ts_secs
+                rescue ArgumentError, TypeError
+                  ts_secs
+                end
+      ts_nanos = begin
+                   Integer ts_nanos
+                 rescue ArgumentError, TypeError
+                   ts_nanos
+                 end
+
+      # Adjust timestamps from the future.
+      #
+      # 1. If the parsed timestamp is later than the end of the year.
```

Comment:
Nit: this isn't a complete sentence. Besides, you only hint at the use case for this adjustment.
It would also be easier to read if the cases were presented in the reverse order.
How about something like:
```ruby
      # Adjust timestamps from the future.
      # There are two cases:
      # 1. The parsed timestamp later in the current year:
      # This can happen when system log lines from previous years are missing
      # the year, so the date parser assumes the current year.
      # We treat these lines as coming from last year.  This could label
      # 2-year-old logs incorrectly, but this probably isn't super important.
      #
      # 2. The parsed timestamp is past the end of the current year:
      # Since the year is different from the current year, this isn't the
      # missing year in system logs.  It is unlikely that users explicitly
      # write logs at a future date.  This could result from an unsynchronized
      # clock on a VM, or some random value being parsed as the timestamp.
      # We reset the timestamp on those lines to the default value and let the
      # downstream API handle it.
```
?

Reply:
Changed.